go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -332,12 +261,99 @@ func newOvnAddressSet(nbClient libovsdbclient.Client, name string, ips []net.IP)
 		// present in the codepath, so no additional handling for that condition has been added here.
 		return nil, fmt.Errorf("failed to create or update address set %+v: %w", addrSet, err)
 	}
-
-	as.uuid = addrSet.UUID
+	as := asf.newOvnAddressSet(addrSet)
 	klog.V(5).Infof("New(%s) with %v", asDetail(as), ips)
 	return as, nil
 }
 
+func (asf *ovnAddressSetFactory) validateDbIDs(dbIDs *libovsdbops.DbObjectIDs) error {
+	for _, key := range dbIDs.GetExpectedExternalIDKeys() {
+		if key == libovsdbops.AddressSetIPFamilyKey {
+			if dbIDs.GetObjectID(key) != "" {
+				return fmt.Errorf("%s key should not be present in db IDs", key)
+			}
+		} else if dbIDs.GetObjectID(key) == "" {
+			return fmt.Errorf("dbIDs expected to have %s key set", key)
+		}
+	}
+	return nil
```

Comment:
Now that I look at this, I would probably encapsulate the logic in DbObjectIDs with methods like `HasKeys(keys) bool`, `HasAllKeys() bool`, `HasAllKeysExcept(keys) bool`. It might be useful in other places.

Reply:
for now addressSetfactory is a very special case, in all the other cases `DbObjectIDs` will be completely filled by one entity, so the verification itself won't make a lot of sense.
So maybe let's wait for some other objects that would need that functionality before actually implementing additional functions?

Reply:
OK.

My comment had two contributing factors:
- not sold on the `GetExpectedExternalIDKeys` name
- the interpretation of what having an empty string value means in regards to object ids should happen inside DB object IDs itself and not outside.

Reply:
i replaced it with `GetUnsetKeys` method, which hopefully should be enough for any verification